src/Concerns/ManagesTransactions.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Jenssegers\Mongodb\Concerns;
+
+use Closure;
+use MongoDB\Client;
+use MongoDB\Driver\Exception\RuntimeException;
+use MongoDB\Driver\Session;
+use function MongoDB\with_transaction;
+
+trait ManagesTransactions
+{
+    protected ?Session $session = null;
+
+    /**
+     * @return Client
+     */
+    abstract public function getMongoClient();
+
+    public function getSession(): ?Session
+    {
+        return $this->session;
+    }
+
+    private function getSessionOrCreate(): Session
+    {
+        if ($this->session === null) {
+            $this->session = $this->getMongoClient()->startSession();
+        }
+
+        return $this->session;
+    }
+
+    private function getSessionOrThrow(): Session
+    {
+        $session = $this->getSession();
+
+        if ($session === null) {
+            throw new RuntimeException('There is no active session.');
+        }
+
+        return $session;
+    }
+
+    /**
+     * Use the existing or create new session and start a transaction in session.
+     *
+     * In version 4.0, MongoDB supports multi-document transactions on replica sets.
+     * In version 4.2, MongoDB introduces distributed transactions, which adds support for multi-document transactions on sharded clusters and incorporates the existing support for multi-document transactions on replica sets.
+     *
+     * @see https://docs.mongodb.com/manual/core/transactions/
+     */
+    public function beginTransaction(array $options = []): void
+    {
+        $this->getSessionOrCreate()->startTransaction($options);
+        $this->transactions = 1;
```

Comment:
Where did this property come from? I don't recall seeing it in the previous PRs and it's not declared in this trait.

As-is, this looks like you're creating a public property at runtime ([deprecated in 8.2](https://php.watch/versions/8.2/dynamic-properties-deprecated)).

Reply:
This is declared in the original `Connection` class: https://github.com/laravel/framework/blob/213e07ef363bb6f2355254a5a9a0da33dcd692ad/src/Illuminate/Database/Connection.php#L127. I've re-declared the property in the trait to make it more explicit, especially since the trait could be used outside of the connection class context.